### DISCUSSION TITLE
Governance Update: self-nominations for the initial group of Jujutsu Maintainers are open!

### DISCUSSION THREAD

**Discussion by thoughtpolice** [jj-vcs/jj#5042] [General]
Howdy *,

After a lot of deliberation and drafting, we've finally published our first draft the **[Jujutsu Governance structure](https://github.com/martinvonz/jj/pull/5040)**! Go check it out! Enjoy the beautiful prose, and please fix my typos. It's almost over... Almost!

While we're open to feedback on the governance structure &mdash; open questions, rule clarifications, general structural refinements &mdash; one key component is something we will begin now: we need to elect an initial group of **Five Maintainers** to jumpstart the whole process and be responsible for decision making.

### What's a Maintainer?

Quoting the shiny `GOVERNANCE.md` draft:

> At the top of the hierarchy are the **Maintainers**. We consider maintainers to
be the people who contribute, review, guide, and collectively make decisions
about the direction and scope of the project.
>
> A typical maintainer is not only someone who has made "large" contributions, but
someone who has shown they are continuously committed to the project and its
community. Some expected responsibilities of maintainers include (but are not
exclusively limited to):
>
> - Displaying a high level of commitment to the project and its community, and
being a role model for others.
> - Writing code &mdash; a lot of code, especially "glue code" or "grunt work";
fixing bugs, improving processes, making judgments on dependencies, handling
security vulnerabilities, and so on and so forth.
> - Reviewing code submitted by others &mdash; with an eye to maintainability,
performance, code quality, and "style" (fitting in with the project).
> - Participating in design discussions, especially with regards to architecture
or long-term vision.
> - When necessary, participate in a collective voting process to resolve
conflicts and make decisions for which there is no clear answer.
>
> This is not an exhaustive list, nor is it intended that every maintainer does
each and every one of these individual tasks to equal amounts. Rather this is
only a guideline for what maintainers are expected to conceptually do.
>
> In short, Maintainers are the outwardly visible stewards of the project.

Note that these rules are NOT all an explicit or exact requirement; if you participate actively but have less patches than other developers, that's OK! We won't be checking all your grades or anything.

### What's the process for picking the initial group?

And after a bunch of deliberation, we've decided on a simple process:

- For the next **three weeks**[^1], we will be taking open self-nominations from any existing contributor
  - Right here in this discussion thread!
  - That would mean nominations **end on December 27th, 2024**.
- Technically, nominations are open, meaning *you* can nominate anyone you like...
  - But in practice we suspect most people will self-select for this initial group.
- At the end of this, the initial set of maintainers will be decided upon by two beloved members of the community in private discussion, who will consist of **the two initial Maintainers**:
  - Martin von Zweigbergk (@martinvonz), and
  - Yuya Nishihara (@yuja)
- A final decision should be made in short order after the nomination period closes, barring holiday parties/vacation, travel shenanigans, etc.

[^1]: Three weeks has been chosen as an arbitrary number, but one important note is that Emily Shaffer is on leave at the moment, and we'd like to wait until she is able to review the governance document, and possibly self-nominate.

To be clear: Yuya and Martin being part of the initial set means that means **there are three open slots available!** Many thanks to them, as they both graciously agreed to take on this burden and decide the initial set. (In case it isn't obvious, Martin and Yuya both are the *de facto* main contributors, reviewers, and maintainers; we felt it was both right and respectful of their labor to allow them this privilege, and we're very happy they accepted.)

### How can you self-nominate?

- Post in this thread
- Introduce yourself; and maybe a sentence or two about what you've done, or want to achieve, or why you think *you've* got what it takes.

That's it!

### What happens next? What if I get elected?

We're hoping that this will proceed independently of any reviews/updates to the governance document and expect it will be a relatively quiet and straightforward process. Once this is done, we'll update the governance document with the names of the chosen few and will proceed under the framework set there. We'll also announce it on all the regular channels.

To be clear: we don't expect the initial set of elected maintainers will need to do anything like weekly meetings, check-ins, etc. This is to be left up to their discretion, so don't worry; things can carry on as normal if you're elected. You won't need to pencil in anything on your schedule.

### Anything else?

That's all from us. Now, let the games begin!

**Comment by thoughtpolice** [comment]
To start the process:

I nominate myself, Austin Seipp (@thoughtpolice). I've been around for a little while now and involved in a bunch of areas from writing code, to helping users, to proselytizing endlessly my coworkers. Thanks for the time and consideration!

**Comment by tim-janik** [comment]
As a follow up, I'd like to nominate @ilyagr who I think is an outstanding contributor to the project.
He is always kind and helpful, patiently helps a newcomers on Github and Discord, goes the extra mile to connect the IRC community and as far as I can tell from my limited perspective is also quite knowledgeable about the source code. His PR feedback also seems to be spot on in the cases I have seen.

**Reply by ilyagr** [reply]
It feels so nice to be nominated! Thank you! In short, yes, I'd be happy to be a maintainer.

I am quite interested in and happy about the community that sprung up around `jj`. I think it's a happy little place, and I hope it can continue to be so for a long time. I hope to foster people helping each other and discussing interesting ideas however I can. In the past, I created our docs website and the IRC bridge with this in mind.

I've contributed to many different parts of `jj`, adding or fixing a bit here and there over the past couple of years. While my code contributions are dwarfed by Martin's or Yuya's, there were quite a few.

**Comment by arxanas** [comment]
I nominate myself, Waleed Khan (@arxanas) as a maintainer.

## Open-source experience

I've been a [jj contributor since 2021](https://github.com/martinvonz/jj/commits?author=arxanas).

I've been working on [git-branchless](https://github.com/arxanas/git-branchless) since 2020, which some readers may be familiar with. (git-branchless is an open-source project with similar goals as jj, but a different approach.)

## Professional experience

I've spent my career working in developer infrastructure and tooling ([LinkedIn](https://www.linkedin.com/in/waleedkhan000)), and I hope to bring specialized expertise to the maintainer group.
- Example: I implemented [jj's Watchman integration](https://martinvonz.github.io/jj/v0.24.0/config/#watchman) specifically to handle large monorepos, based on [relevant professional experience](https://blog.waleedkhan.name/incremental-watchman/).

Among other things, I have worked on or adjacent to monorepo-scale source control tooling, including at companies like Meta and Twitter.
- Example: See [Build-aware sparse checkouts](https://blog.waleedkhan.name/build-aware-sparse-checkouts/) for monorepo tooling I worked on at Twitter.

> [!NOTE]
> At present (2024-12-16), my involvement in jj is personal, and does not represent any corporate interest. 

## Vision and goals

I'm passionate about improving the UX for developer tooling for novices and experts alike.

There are two main aspects of my long-term vision for jj:
- Novices and intermediate users should find the interface intuitive, predictable, and easy to experiment with.
  - Example: See [git undo: We can do better](https://blog.waleedkhan.name/git-undo/#motivation) for my thesis on how improving the developer feedback loop benefits novice users.
- Experts and power users should find the interface highly flexible and able to streamline common workflows.
  - Example: See git-branchless's [`git test`](https://github.com/arxanas/git-branchless/wiki/Command:-git-test) for a workflow I'd like to bring to jj one day.

In doing so, I hope to bring unique technical and UX perspective to jj based on the relevant research. Some examples of the research I draw on:
- My original research:
  - [Where are my Git UI features from the future? (2023)](https://blog.waleedkhan.name/git-ui-features/)
  - [Patch terminology (2024)](https://blog.waleedkhan.name/patch-terminology/)
- Git UX research:
  - [Gitless (2013–2016)](https://gitless.com/#research)
  - [Investigating & Archiving the Scholarly Git Experience (2019–2022)](https://investigating-archiving-git.gitlab.io/)
- Other source-control-related research:
  - [Keeping master green at scale (2019)](https://www.uber.com/blog/research/keeping-master-green-at-scale/)
  - [Flake aware culprit finding (2021)](https://research.google/pubs/flake-aware-culprit-finding-2/)
  - [Lazy Merging: From a Potential of Universes to a Universe of Potentials (2023)](https://eceasst.org/index.php/eceasst/article/view/2318)

**Comment by PhilipMetzger** [comment]
Hello I'd like to nominate myself @PhilipMetzger as a maintainer. 

I've been around for a bit longer than Austin and visited Martin and Ilya in Mountain View last year. I also have been involved in multiple designs such as `jj run`, `sparse-v2`, `jj api` and more. I also have been showing up for the last two years, where I either helped users get accustomed to `jj` in the Discord or in the project with documentation fix-ups. I also had some controversy in Spring, so take it as you will. 

I'm also happy if the three mentioned above get in, even if that leaves no space for a EMEA representative.

**Comment by thoughtpolice** [comment]
Hello folks, I just wanted to give an update that the voting period is now CLOSED. Thank you to everyone who applied; @martinvonz and @yuja will take it from here and come back with the initial set of maintainers soon, probably sometime early in the New Year. Thanks again!

**Comment by martinvonz** [comment]
Thanks to everyone who offered to become a maintainer! Yuya and I discussed the nominations and agreed on @thoughtpolice, @ilyagr, and @arxanas. I'll invite you as members of the Maintainers group in the jj-vcs org, and as co-owners of the crates on crates.io.
